cpp/src/gandiva/function_registry_math_ops.cc

```diff
@@ -0,0 +1,86 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gandiva/function_registry_math_ops.h"
+#include <utility>
+#include <vector>
+
+namespace gandiva {
+
+void FunctionRegistryMathOps::GetMathOpsFnSignature(SignatureMap* map) {
+  // list of registered native functions.
+
+  static NativeFunction math_fn_registry_[] = {
+      // extended math ops
+      UNARY_SAFE_NULL_IF_NULL(cbrt, int32, float64),
```

Comment:
When there's repetition like this, macros. I want to see this:

```C++
MATH_UNARY_OPS(cbrt),
MATH_UNARY_OPS(exp),
MATH_UNARY_OPS(log),
...
```

Reply:
changed as per suggestion.